### PR TITLE
pr-upload: ensure bottle and formula version match

### PR DESCRIPTION
**Pull request by jonchang**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Part of https://github.com/Homebrew/brew/issues/7917. Ensure that the bottle being uploaded (as described by its JSON file) is the same version as the formula it's being uploaded for.

### CODE REVIEW

File: Library/Homebrew/dev-cmd/pr-upload.rb [hunk 1]
```diff
@@ -32,26 +32,40 @@ def pr_upload_args
...
     bottle_args << "--debug" if args.debug?
     bottle_args << "--keep-old" if args.keep_old?
     bottle_args << "--root-url=#{args.root_url}" if args.root_url
-    odie "No JSON files found in the current working directory" if Dir["*.json"].empty?
-    bottle_args += Dir["*.json"]
+    bottle_args += json_files
 
     if args.dry_run?
       puts "brew #{bottle_args.join " "}"
       puts "Upload bottles described by these JSON files to Bintray:\n  #{Dir["*.json"].join("\n  ")}"
```
Comment (dawidd6) [inline Library/Homebrew/dev-cmd/pr-upload.rb:64]:
```suggestion
      puts "Upload bottles described by these JSON files to Bintray:\n  #{json_files.join("\n  ")}"
```

Comment (jonchang) [reply Library/Homebrew/dev-cmd/pr-upload.rb]: Good catch! :eyes:

File: Library/Homebrew/dev-cmd/pr-upload.rb [hunk 2]
```diff
@@ -32,26 +32,40 @@ def pr_upload_args
     end
   end
 
+  def check_bottled_formulae(json_files)
+    json_files.reduce({}) { |hash, json| hash.deep_merge(JSON.parse(IO.read(json))) }.each do |name, hash|
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/pr-upload.rb:36]:
```suggestion
    json_files.reduce({}) do |hash, json| 
      hash.deep_merge(JSON.parse(IO.read(json)))
    end.each do |name, hash|
```

File: Library/Homebrew/dev-cmd/pr-upload.rb [hunk 3]
```diff
@@ -32,26 +32,40 @@ def pr_upload_args
...
+      formula_path = HOMEBREW_REPOSITORY/hash["formula"]["path"]
+      formula_version = Formulary::FormulaLoader.new(name, formula_path).get_formula("stable").version
+      bottle_version = Version.new hash["formula"]["pkg_version"]
+      if formula_version != bottle_version
+        odie "Bottles are for #{name} #{bottle_version} but formula is version #{formula_version}!"
+      end
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/pr-upload.rb:42]:
```suggestion
      next if formula_version == bottle_version

      odie "Bottles are for #{name} #{bottle_version} but formula is version #{formula_version}!"
```

**Pull request merged**